feat: Improve month handling for Revenue Analytics

rafaeelaudibert opened the pull request:
We're now properly using the dates passed in as `dateRange` for our revenue queries.

Frontend was also updated to present a more sensible set of defaults for our timeframe picker - removed the interval because we wanna control it ourselves.

![image](https://github.com/user-attachments/assets/247bad10-73d5-4626-9e94-208bc48105aa)

We still need to hide some of the charts depending on the query (for example, if we query just one month, then we shouldn't display growth, for example), and also some height-related stuff (line chart is not rendering tall enough in some situations, see image above)

Closes #31262
Closes #31255

rafaeelaudibert committed:
feat: Improve month handling for revenue

We're now properly using the dates passed in as `dateRange` for our revenue queries.

Frontend was also updated to present a more sensible set of defaults for our timeframe picker - removed the interval because we wanna control it ourselves.

We still need to avoid querying *some* data based on the date selection AND properly adding a toggle between `month` and `all` for the top customers feature - toggle between a table and a chart view.

rafaeelaudibert committed:
feat: Allow growth/top customers to toggle between line/table

lricoy commented on products/revenue_analytics/frontend/RevenueAnalyticsFilters.tsx:
```diff
@@ -0,0 +1,68 @@
+    },
+    {
+        key: 'Year to date',
+        values: ['yStart'],
+        getFormattedDate: (date: dayjs.Dayjs): string => formatDateRange(date.startOf('y'), date.endOf('d')),
+        defaultInterval: 'month',
```
Somewhat related, but more of an overall question than this PR: can we ingest events in the future? I've seen people do some weird stuff on financial reports, so maybe it is something you'll want to take a look

rafaeelaudibert replied:
Ah, yes, we can, we dont validate the timestamp. And that's fine, really. I'll update this to make that behavior slightly better

lricoy commented on products/revenue_analytics/frontend/RevenueAnalyticsScene.tsx:
```diff
@@ -75,32 +73,18 @@ export function RevenueAnalyticsScene(): JSX.Element {
             <OverviewTile />
-
             <GrossRevenueTile />
-            <RevenueGrowthRateTile />
-            <RevenueChurnTile />
+
+            <LemonDivider className="mt-6" />
+
+            <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
+                <RevenueGrowthRateTile />
+                <TopCustomersTile />
+            </div>
```
Robbie school of Tiles🥇

lricoy commented on products/revenue_analytics/frontend/revenueAnalyticsLogic.ts:
```diff
@@ -139,32 +203,25 @@ export const revenueAnalyticsLogic = kea<revenueAnalyticsLogicType>([
-                    [RevenueAnalyticsQuery.TOP_CUSTOMERS]: {
-                        kind: NodeKind.DataTableNode,
-                        source: {
+                    [RevenueAnalyticsQuery.REVENUE_GROWTH_RATE]: wrapWithDataTableNodeIfNeeded(
```
Oh, the wrapping here is clever!

lricoy reviewed (approved):
Seems solid for testing it in prod to make sure the assumptions are correct, congrats! Don't want to block you, just keep in mind the static date fallback comment as it may influence other stuff. 

I also did not run everything locally, so if you want extra checks, wait for later today :) 

>also some height-related stuff (line chart is not rendering tall enough in some situations, see image above)

I am not sure if that's what you mean, but I think I faced this on Page Reports and I was missing some `{ embedded: true }` or other insightProp to make it look better.

rafaeelaudibert commented on posthog/queries/util.py:
```diff
@@ -99,10 +97,10 @@ def get_earliest_timestamp(team_id: int) -> datetime:
         return results[0][0]
-    else:
-        return timezone.now() - DEFAULT_EARLIEST_TIME_DELTA
+    return datetime.fromisoformat(EARLIEST_TIMESTAMP)
```
Ah yes, I have to ask the team about this. It's a 4-year old assumption, I *think* we can change this safely, but will check

rafaeelaudibert committed:
refactor: Use fixed fallback for revenue query data

When running a query with the `all` timeframe, we usually run a query to figure out when the earliest ingested event happened. That doesn't make any sense for data warehouse tables, so let's change this slightly.

For simplicity, let's simply assume January 1st 2015 for now. Some people might ingest historic data, but we can cross that bridge once and if we get there

rafaeelaudibert committed:
refactor: Remove commented out code

rafaeelaudibert committed:
fix: Add new property to other `QueryDateRange`

We have 2 of these, yuck

rafaeelaudibert committed:
feat: Add stories for RevenueAnalytics

rafaeelaudibert committed:
feat: Make charts have some implicit height

rafaeelaudibert committed:
refactor: We don't need this, just one month is rendered and that's all right

rafaeelaudibert committed:
refactor: Improve limit in the backend

rafaeelaudibert committed:
fix: Export type

rafaeelaudibert committed:
refactor: Add notice on what `customer_id` is

rafaeelaudibert committed:
refactor: Improve overview layout

New layout is much more sane, and adapts to smaller/bigger screens much better

rafaeelaudibert committed:
fix: Fix storybook

rafaeelaudibert committed:
feat: Improve loading behavior for revenue analytics

rafaeelaudibert committed:
feat: Add `Onboarding` view to Revenue Analytics snapshot

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

rafaeelaudibert committed:
refactor: Improve check for `ProductIntroduction`

CI is generating some weird UI including both `ProductIntroduction` AND the tables, let's change this and avoid using `isEmpty`

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

rafaeelaudibert committed:
fix: Regenerate schema files
